testsuite/features/step_definitions/common_steps.rb

```diff
@@ -70,18 +70,24 @@
   step %(I wait at most #{DEFAULT_TIMEOUT} seconds until event "#{event}" is completed)
 end
 
-When(/^I wait at most (\d+) seconds until event "([^"]*)" is completed$/) do |final_timeout, event|
+When(/^I wait at most (\d+) seconds until event "([^"]*)" is completed$/) do |timeout, event|
+  starting = Time.now
   # The code below is not perfect because there might be other events with the
   # same name in the events history - however, that's the best we have so far.
   steps %(
     When I follow "Events"
     And I follow "Pending"
-    And I wait until I do not see "#{event}" text, refreshing the page
+    And I wait at most #{timeout} seconds until I do not see "#{event}" text, refreshing the page
     And I follow "History"
     And I wait until I see "System History" text
     And I wait until I see "#{event}" text, refreshing the page
+  )
+  ending = Time.now
```

Comment:
I'd avoid this instantiation, and use Time.now directly on the next line.

Reply:
code more readable...

Reply:
ok